tests/unit/unit-util.cpp

```diff
@@ -156,6 +156,64 @@ TEST_P(UnitUtilIntTests, Oracles)
   }
 }
 
+TEST_P(UnitUtilTests, cnf_to_dimacs)
+{
+  if (s->get_solver_enum() != CVC4)
+  {
+    return;
+  }
```

Comment:
I think we do want to make sure this utility can be used outside of `cvc4` and also tested.
So please try to do something similar to this: 
https://github.com/makaimann/smt-switch/blob/master/tests/test-logging-solver.cpp

They only create Logging Solvers. These are guaranteed not to change the structure of the input formulas. And they do this for every supported solver.
You might want to create a separate class in this file for the dimacs tests, and for that class, iterate (using the google tests mechanism) over only logging solvers.

Reply:
Should I transfer my test from the earlier file to this file?

Reply:
I don't think I exactly understood what I should do

Reply:
Good question!
Your current test file ends with these two commands:
https://github.com/makaimann/smt-switch/pull/250/files#diff-77a1d128c2d18757a0ac996385ff8f6ebc896eb2abf3c83647bdf1ee35124d3dR218
https://github.com/makaimann/smt-switch/pull/250/files#diff-77a1d128c2d18757a0ac996385ff8f6ebc896eb2abf3c83647bdf1ee35124d3dR222

The former controls the executions of the tests that are defined with `UnitUtilTests`, and the latter relates to `UnitUtilIntTests`.
In the former, the command tells google test to execute all tests related to `UnitUtilTests` on all solvers that support `TERMITER`.
In the latter, a more restricted set of solvers is invoked -- those that support both `TERMITER` and `THEORY_INT`.

Now, `UnitUtilTests` and `UnitUtilIntTests` are defined [here](https://github.com/makaimann/smt-switch/pull/250/files#diff-77a1d128c2d18757a0ac996385ff8f6ebc896eb2abf3c83647bdf1ee35124d3dR31) and [here](https://github.com/makaimann/smt-switch/pull/250/files#diff-77a1d128c2d18757a0ac996385ff8f6ebc896eb2abf3c83647bdf1ee35124d3dR50).

Then, each test is associated with one of them. For example, [this](https://github.com/makaimann/smt-switch/pull/250/files#diff-77a1d128c2d18757a0ac996385ff8f6ebc896eb2abf3c83647bdf1ee35124d3dR85) test is associated with the former.

What I am suggesting is to add a third class, call it `UnitUtilDimacsTests`. It might contain only one test, but that's OK. Maybe we will add more tests to it later.
So your current test should be associated with this new class. 
Then, you can add a line at the end of the file, just like [this](https://github.com/makaimann/smt-switch/pull/250/files#diff-77a1d128c2d18757a0ac996385ff8f6ebc896eb2abf3c83647bdf1ee35124d3dR218) line.

But, instead of using `filter_solver_configurations`, use `filter_solver_enums`. A `SolverConfiguration` is a solver enum + a flag that states whether it is a logging solver or not. For this test, we only want logging solvers. So we just range over the enums.
Then, in your new `UnitUtilDimacsTests` class, the setup line will not be like it is now:
https://github.com/makaimann/smt-switch/pull/250/files#diff-77a1d128c2d18757a0ac996385ff8f6ebc896eb2abf3c83647bdf1ee35124d3dR37

Because, now, `GetParams` will return a `SolverEnum`, while `create_solver` expects a `SolverConfiguration`.
So create a `SolverConfiguration` that takes the enum, and give it ` true` flag for the `logging` argument. This will ensure you will get a logging solver, which in turn, will ensure that the original assertions are not rewritten before you process them.

This was a long answer, I hope it is not too confusing :) Let me know if you have more questions. There might be an easier way to achieve the same goal of course, but this is how I would do it.

Reply:
Thanks a lot! Gained a lot of clarity.